diff --git a/babel-plugin.js b/babel-plugin.js
index 06655e293439957fe7432e087ac988a8b6cf8fb2..c1f884317826745eb52d4bdafeb567c8afffe8f3 100644
--- a/babel-plugin.js
+++ b/babel-plugin.js
@@ -658,13 +658,13 @@ function transformFactory(t, path, state, factoriesUsed, debugSids, addLoc, addN
   });
   const sid = generateStableID(state.file.opts.root, state.filename, resultName, loc.line, loc.column, debugSids);
   const factoryConfig = {
-    SID: JSON.stringify(sid),
+    SID: t.stringLiteral(sid),
     FN: node,
     FACTORY: withFactoryImportName
   };
   if (addLoc || addNames) {
-    factoryConfig.NAME = JSON.stringify(!resultName || resultName === '' ? 'none' : resultName);
-    factoryConfig.METHOD = JSON.stringify(importedName);
+    factoryConfig.NAME = t.stringLiteral(!resultName || resultName === '' ? 'none' : resultName);
+    factoryConfig.METHOD = t.stringLiteral(importedName);
   }
   if (addLoc) {
     factoryConfig.LOC = makeTrace(state.fileNameIdentifier, loc.line, loc.column, t);
